### ISSUE TITLE
reward system for this github fork

### ISSUE THREAD

**Issue by jjhesk** [ccxt/ccxt#1302]
it is just a suggest for better development for this fork repo. The situation is that you are taking all the work to support all exchanges api and it has slow progress and it seems to be a bottle neck issue. Lets think of a way that the development support can be scalable for every developer with measurable reward.

**Comment by jjhesk** [comment]
Suggested by the latest avocation. There is a IBO to manage the growth of a project with a self-driven ego system. check out https://cointelegraph.com/explained/how-initial-bounty-offering-can-help-the-unbanked-explained

**Comment by jjhesk** [comment]
I think this "rating feature that network will able to guide and lower their transactional risk for every single trade." should be discussed with another open discussion for the development roadmap.

**Comment by jjhesk** [comment]
developer enrolment procedure: should be qualified with 3 initial and successful merges to the main chain will be qualified for reward system to gain future PRs. (requirements: compiled successfully with the test runner and has been all the functions api implemented to the file for the new exchange)

manager enrolment procedures: should be qualified with 3 initial voted contributions or initiatives that drive for future growth of this project. e.g. successfully discover and documented new exchanges, discover and confirm the peoples bug report and possible point out the line of code that have issue at for the extra bonus coin reward.

**Comment by kroitor** [comment]
Oh, wow! Thx for the proposal! Sounds very cool! )

One question though: who pays for it? ) 

During last 6 months our users have sent approximately $200 worth of crypto donations in total. It isn't a secret and you can see the transaction history on those addresses. The question is: how do you develop a system for the IBO with initial funding close to zero? ) 

In other words, your proposal sounds so great, but it requires a significant amount of work just to develop a system for the IBO. And while are all for it, to be realistic, we don't see anyone ready to pay for that work. People mostly want it for free or for a "thank you". Also, we have an open collective started here: https://opencollective.com/ccxt. It has been started like a month ago. Since then we got *a crowd* of 0 people who have sent something, and we have collected a *huge amount* of 0 money (approximately). 

What would you say if I send you, say, a half of the zero total money collected, for you to develop an IBO system for this project? ;)) In other words, how is this **realistic** ? Do you have a plan (with timing) what should be done step by step in order to reach the endgoal of making it a self-sustained project? Because if you try to make one, you will inevitably end up facing the money question. WHat do you think on this? ) Thx!

**Comment by jjhesk** [comment]
1. i think we should talk to the exchanges for the funding. Thats my first thought. They are normally cash cow from the market. They will totally support this kind of project because it helps them to reach bigger market goal.

**Comment by kroitor** [comment]
> They will totally support this kind of project because it helps them to reach bigger market goal.

Are you sure about this? Check this out: https://gist.github.com/kroitor/67cf3fd9b607d1bb2cbcc7b35b49faa6
Looks like not all of them want it at all. Mostly the smaller exchanges want to be added to the library, but they don't want to pay anything for it, they just want a free chance to compete with the giants. And the giants don't seem to care at all...

**Comment by jjhesk** [comment]
for smaller exchanges should have lower priority for the api to be added. there should a fine line that cover when the api should be added for free and when its not. they will need to try hard from their own marketing in order to get the volume running. also we can be the watch dog among all exchanges, this valuable because this many trader will need this information for their funding direction indications.

**Comment by kroitor** [comment]
@jjhesk lower priorities are ok, but have you checked out that link above? It's about BitMEX, that is the 5th topmost exchange, not a small one.

**Comment by jjhesk** [comment]
yes, im expecting that the likelyhood to collect the fee from the exchange will happen in the future as these kind of exchanges will be disrupted by the descentralized exchanges.

**Comment by kroitor** [comment]
> i think we should talk to the exchanges for the funding.

BTW, most of them don't even bother to reply. And it's a problem. We have established pretty good connections with some exchanges and their support/tech teams and no connections with others, because they never reply at all.

**Comment by jjhesk** [comment]
yes, this is why we need to create an incentive for the exchanges to come to us because our position is to holding the longest history of exchange api records. We should able to develop an eco self check system for the health of these exchanges and their integrity. For this point itself creates the business model. funding will be easy to acquire.

**Comment by kroitor** [comment]
> We should able to develop an eco self check system for the health of these exchanges and their integrity

Alright, but who pays for the initial development work? ) That's a request of a serious infrastructure, that would require a fews months of dedicated work, how do you fund it **now** ?

**Comment by jjhesk** [comment]
no worry, funding should be quickly and sent to the team. i go find them funding.

**Comment by kroitor** [comment]
@jjhesk 

> i go find them funding.

You're amazing! ) Thank you so much, I really wish it will help us continue this whole effort! )

**Comment by jjhesk** [comment]
this is the plan for mine maybe, sending out email for marketing. having these guy to talk about the team. once you have done some of these the fund will come knock on the door.
1. dimitar@thenextweb.com / https://thenextweb.com/author/dimitarmihov/
2. https://www.crunchbase.com/person/natasha-lomas / https://techcrunch.com/author/natasha-lomas/

**Comment by kroitor** [comment]
Sounds like a plan indeed! )

> dont forget to give me some credit when you guys are going to the Moon! =)

Sure, we will! We actually hope to build a bigger team from our community. We won't forget anything, believe me )

Thank you!

**Comment by wannesdemaeght** [comment]
If I could chime in...
One way to get some more revenue out of this excellent work you're doing, would be to add subscription-based services.

One area that immediately pops into my mind is the websockets, even for exchanges that don't support websockets.
Basically you could run a VPS with REST ticker input for all exchanges, and output that stream constantly via websocket to subscribed users.
It would save us the API calls, and you could get a nice revenue from it.

Hell, I'd pay for it.

**Comment by kroitor** [comment]
@wannesdemaeght thanks for your thoughts on this! We also considered doing this type of service, however,  launching such a reliable and trustworthy aggregator would take at least a few weeks of work. Along with other urgent tasks it can take much more time. But we will do our best to make it, of course, and we also have several more ideas on microservices, which we think would be very useful. Our topmost difficulty now is the lack of time.

**Comment by mkutny** [comment]
> We have noticed a huge increase in API abuse recently, and it nearly always has ccxt in headers. Could you just remove BitMEX support from your library?

OMG, I can't believe it! ))))))
OMG!!! )))))))))

**Comment by kroitor** [comment]
@mkutny yeah, I was shocked too ))

**Comment by mkutny** [comment]
But, well, you said exchanges don't wanna pay )

You know, what? )))

Put `User-Agent` back and ask them to pay you for 'removing' support from the library )))

**Comment by kroitor** [comment]
@mkutny also thought about that ) However, I don't think many of them would be happy to pay then )))

**Comment by kroitor** [comment]
@mkutny you know what's even more important, bitmex admins sent me their production ips and the user that I should not have access to, in the first place. And that in itself is making it even easier to abuse them, in fact. Even if I removed bitmex from ccxt, it would not fix anything for them, because they were abused by users with older versions. Anyway, it kinda illustrates the quality of their service.

**Comment by kroitor** [comment]
Good news, everyone!

![good-news-everyone](https://user-images.githubusercontent.com/1294454/35069860-7f3835be-fbec-11e7-92a6-956e197e8e50.jpg)

Today we got our first backer in the opencollective! ) We don't know who this person is (anonymous), but just in case he or she might be reading this, **thanks a lot**! )

**Comment by jjhesk** [comment]
@kroitor =) as i mention. please dont worry about backers. Money is not problem. The very important thing is to keep this lib alive and accessible to developers.

**Comment by jjhesk** [comment]
Whats in mind right now is to issue coins just for this project. A project like this should be tradable in the market. For a project like right is totally undervalue.

**Comment by jjhesk** [comment]
@kroitor i think we can introduce https://gitcoin.co/ to start. See what we can do with this program.

**Comment by jjhesk** [comment]
@kroitor is that a way to contact u in email?
